items.js

```diff
@@ -0,0 +1,90 @@
+const cl = console.log;
+
+const enhTree = {
+  0: "",
+  1: "[+1] ",
+  2: "[+2] ",
+  3: "[+3] ",
+  4: "[+4] ",
+  5: "[+5] ",
+  6: "[+6] ",
+  7: "[+7] ",
+  8: "[+8] ",
+  9: "[+9] ",
+  10: "[+10] ",
+  11: "[+11] ",
+  12: "[+12] ",
+  13: "[+13] ",
+  14: "[+14] ",
+  15: "[+15] ",
+  16: "[PRI] ",
+  17: "[DUO] ",
+  18: "[TRI] ",
+  19: "[TET] ",
+  20: "[PEN] "
+};
+
+const success = item => {
+  const modItem = Object.assign({}, item);
+  if (modItem.enhancement < 20) {
+    modItem.enhancement += 1;
+  }
+  setName(modItem);
+  return modItem;
+};
+
+const fail = item => {
+  const modItem = Object.assign({}, item);
+  const lowLimit = modItem.type === "armor" ? 5 : 7;
+  const enhancement = modItem.enhancement;
+  if (enhancement <= lowLimit) {
+    setName(modItem);
+    return modItem;
+  }
+  if (enhancement < 15) {
```

Comment:
For something like this, a switch statement might be easier to read. https://stackoverflow.com/questions/6665997/switch-statement-for-greater-than-less-than
performance might drop a little though.